### ISSUE TITLE
[FR] Customise expiration on email links

### ISSUE THREAD

**Issue by antpuleo2586** [firebase/firebase-admin-node#697]
**Is your feature request related to a problem? Please describe.**
We need to be able to set the expiration time when generating an email link from our backend, because some users take some time before they actually click on the link.

**Describe the solution you'd like**
Is it possible to customise the expiration time, or at least have some visibility into what the current expiration time for these email links is?

**Describe alternatives you've considered**
Asking the user to re-request the email link, but this is not a great user experience, to click on an email link only to have to request another email link from within the mobile app.

**Additional context**
Using React Native with firebase auth email links & dynamic links.

**Comment by rsgowman** [comment]
This is largely a duplicate of https://github.com/firebase/firebase-js-sdk/issues/1884.

tl;dr: Unfortunately, it's not currently possible to set the expiration time for any of the generated email links. And unfortunately, we don't publish the expiration times either. (It's subject to change.) Sorry! :(

However, it's relatively safe to assume that the email verification links will expire after a small number of days, while the rest will expire after a number of hours.

If you want this to be changed, you could file a feature request (https://firebase.google.com/support).

**Comment by bsplosion** [comment]
@rsgowman why is there not a published expiration?  I can understand it being subject to change, but specifications are, and they generally are fine being published.  Merely saying it's safe to say they "expire after a number of hours" is not helpful - it could be 1 hour, 10 hours, 6247 hours, any number.  Has this stance changed at all in the past couple of years?  You're the only auth provider I could find which fails to document these basics of user management.

**Comment by rsgowman** [comment]
I haven't been involved here for a few years, so I'm not sure if anything's changed or not. @egilmorez might know though.

**Comment by Hoppi164** [comment]
@egilmorez  
Do you know if there has been any update on this?
I've searched for this information across firebase docs but not been able to find anything..

It's rather important to know how long until the email verification link expires.

**Comment by tan1232212** [comment]
why they closed this issue LOL

**Comment by earok** [comment]
I need a solution for this. A client is complaining that the time it takes for a password reset email to get through their email quarantine system means that the password reset link is expired before they can click the link.

**Comment by okevino47** [comment]
Does someone found a solution ? i am on the exact same problem.
Thank you folks

**Comment by Hoppi164** [comment]
@okevino47 

Nope; and i doubt they ever will fix it / publish the specifications.

I just use the links and hope for the best,
We designed a UI that attempts to log the user in with the link, and if it fails we prompt them to send the email again.
